firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -1775,19 +1775,25 @@ class BrowserViewController: UIViewController,
         }
     }
 
+    func lockIconImageName(for tab: Tab?) -> String? {
+        guard let tab, let hasSecureContent = tab.webView?.hasOnlySecureContent else { return nil }
+
+        let lockIconImageName = hasSecureContent ?
+        StandardImageIdentifiers.Large.lockFill :
+        StandardImageIdentifiers.Large.lockSlashFill
+
+        return tab.url?.isReaderModeURL == false ? lockIconImageName : nil
```

Comment:
Why are we removing the lock icon here?

Reply:
When reader mode is active, the lock icon is hidden.

Reply:
Can you point me to where this rule is defined? Ideally in Figma.

Reply:
I've tried to keep the same behavior as before, otherwise it would show the webpage as unsecure.